protocol/encryption/persistence.go

```diff
@@ -766,63 +767,157 @@ func (s *sqlitePersistence) GetHashRatchetKeyByID(groupID []byte, keyID uint32,
 	var key, hash []byte
 	var seqNoPtr *uint32
 
-	err = stmt.QueryRow(groupID, keyID, seqNo).Scan(&key, &seqNoPtr, &hash)
+	oldFormat := ratchet.IsOldFormat()
+	if oldFormat {
```

Comment:
In the flow: `DecryptPayload` -> `decryptWithHR` -> `GetHashRatchetKeyByID` it will be always true, even if the header indicates new format. That's because `ratchet.Key` is nil. Is that expected?

Reply:
mmh, not good :), it works fine since we maintain compatibility, but we should prefer the new key format, i'll fix it

Reply:
wait, is it, isOldFormat 
```
func (h *HashRatchetKeyCompatibility) IsOldFormat() bool {
	return len(h.keyID) == 0 && len(h.Key) == 0
}
```
so keyid would have to be empty for it to be in the old format, but we do set it

```

		ratchet := &HashRatchetKeyCompatibility{
			GroupID: header.GroupId,
			// NOTE: this would be nil in the old format
			keyID: header.KeyId,
		}

```
so should work, or am I missing something?